flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -607,6 +607,195 @@ class TableEnvironmentTest {
     checkData(expectedResult.iterator(), tableResult.collect())
   }
 
+  @Test
+  def testAlterTableModifyColumn(): Unit = {
+    val statement =
```

Comment:
nit: These negative cases is much same in the `table.q`. I think we can keep one...